disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/internal/storage/Storage.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.internal.storage;
+
+import io.opentelemetry.contrib.disk.buffering.internal.storage.files.ReadableFile;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.files.WritableFile;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.responses.ReadableResult;
+import io.opentelemetry.contrib.disk.buffering.internal.storage.responses.WritableResult;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+import javax.annotation.Nullable;
+
+public final class Storage implements Closeable {
+  private final FolderManager folderManager;
+  @Nullable private WritableFile writableFile;
+  @Nullable private ReadableFile readableFile;
+  private static final int MAX_ATTEMPTS = 3;
+  private final AtomicBoolean isClosed = new AtomicBoolean(false);
+
+  public Storage(FolderManager folderManager) {
+    this.folderManager = folderManager;
+  }
+
+  /**
+   * Attempts to write a line into a writable file.
```

Comment:
`line` feels a bit misleading as far as I can tell it's just binary data

Reply:
It's removed now.